kura/org.eclipse.kura.core.crypto/src/main/java/org/eclipse/kura/core/crypto/CryptoServiceImpl.java

```diff
@@ -64,6 +71,30 @@ public class CryptoServiceImpl implements CryptoService {
     private final SecureRandom random = new SecureRandom();
     private SystemService systemService;
 
+    private CharsetEncoder utf8Encoder;
```

Comment:
Please see that CharsetEncoder is not thread-safe: https://docs.oracle.com/javase/7/docs/api/java/nio/charset/CharsetEncoder.html

Reply:
i'll synchronize the methods using them.

Reply:
Is it better to make the methods synchronized or using a dedicated encoder/decoder for each encrypt/decrypt operation?

Reply:
> Is it better to make the methods synchronized or using a dedicated encoder/decoder for each encrypt/decrypt operation?
I don't know, it seems too expensive to me to create an encoder/decoder every time.

Reply:
It doesn't seem that expensive looking at openjdk impl https://github.com/openjdk/jdk/blob/jdk8-b120/jdk/src/share/classes/sun/nio/cs/UTF_8.java#L57 https://github.com/openjdk/jdk/blob/jdk8-b120/jdk/src/share/classes/java/nio/charset/Charset-X-Coder.java.template